ONBOARDING: Working realm functionality

priettt opened the pull request:
If the realm database is empty, characters are loaded from the server. Otherwise, they are directly loaded from the database itself. To load the characters into the database, the user needs to press the refresh button.

Commit:
ONBOARDING: Working realm functionality

If the realm database is empty, characters are loaded from the server. Otherwise, they are directly loaded from the database itself. To load the characters into the database, the user needs to press the refresh button.

fanky10 commented on app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt:
```diff
@@ -21,87 +21,84 @@ const val COMIC_URL_TAG = "comiclink"
 
-    private fun refresh() {
+    private fun listenToRefresh() {
         RxBus.subscribe(view.activity, object : OnRefreshPressedBusObserver() {
```
now that you know RXAndroid you can move away from RxBus

priettt replied:
done!

Commit:
Migrated to RxAndroid, new loading spinner

Replace RxBus uses with RxAndroid. Added a new loading spinner in the middle of the screen.

Commit:
Fixed unit tests

Commit:
Added tests

SartiMau reviewed (changes requested):
Please put a screenshot  showing that all the tests passed

priettt commented:
![screen shot 2018-12-21 at 11 44 09](https://user-images.githubusercontent.com/26326883/50347929-17e7d700-0516-11e9-9178-6558ca4eebf0.png)

fanky10 reviewed (approved):
LG2M
